yasnippet.el

```diff
@@ -3367,7 +3371,6 @@ If so cleans up the whole snippet up."
                      (not (and active-field (yas--field-contains-point-p active-field))))
                  (setq snippets-left (delete snippet snippets-left))
                  (setf (yas--snippet-force-exit snippet) nil)
-                 (setq snippet-exit-hook yas-after-exit-snippet-hook)
```

Comment:
This drops the code that gets the snippet-local binding of this hook (thus the failing test). Consider keeping the `snippet-exit-hook` variable and then let-binding `yas-after-exit-snippet-hook` to its value around the `yas--safely-run-hook` instead.